### PR TITLE
feat(config): load new configs dynamically through rpc

### PR DESCRIPTION
**Pull request by magicalne**

### CODE REVIEW

File: crates/rpc-server/src/registry.rs [hunk 1]
```diff
@@ -441,7 +444,9 @@ impl RequestSubmitter {
                 let state = snap.state().expect("get mem state");
                 let kind = req.kind();
                 let hash = req.hash();
-                match self.req_to_entry(req, &state, queue.len()) {
+                let dynamic_config_manager = &self.dynamic_config_manager.read().await;
+                let fee_config = dynamic_config_manager.get_fee_config();
+                match req_to_entry(fee_config, self.generator.clone(), req, &state, queue.len()) {
```
Comment (Flouse) [inline crates/rpc-server/src/registry.rs:449]: In this background loop,
we should consider the time comsumed by `generator.clone()`.

File: crates/rpc-server/src/registry.rs [hunk 2]
```diff
@@ -457,7 +462,9 @@ impl RequestSubmitter {
                 while let Ok(req) = self.submit_rx.try_recv() {
                     let kind = req.kind();
                     let hash = req.hash();
-                    match self.req_to_entry(req, &state, queue.len()) {
+
+                    match req_to_entry(fee_config, self.generator.clone(), req, &state, queue.len())
```
Comment (Flouse) [inline crates/rpc-server/src/registry.rs]: This operation is in the high frequency loop.

File: crates/dynamic-config/src/manager.rs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+// So that we don't need to restart to take effect every time.
+#[derive(Default)]
+pub struct DynamicConfigManager {
+    config_github_url: Option<String>,
+    fee_manager: FeeConfigManager,
+    whitelist_manager: WhilteListConfigManager,
+}
+
+impl DynamicConfigManager {
+    pub fn create(config: Config) -> Self {
```
Comment (jjyr) [inline crates/dynamic-config/src/manager.rs:24]: We should create new structure DynamicConfig or `RemoteConfig` to only include the configurations that can can be dynamically reloaded.

Use Config structure introcing too much complexity.

**Review by jjyr — CHANGES_REQUESTED** [review]

File: crates/rpc-server/src/registry.rs (hunk 1 above)
Comment (magicalne) [reply crates/rpc-server/src/registry.rs:449]: That should be fine. This clone is an Arc clone. And we only read backends here which is fast. We can consider cache backends in the future maybe.

**Review by jjyr — APPROVED** [review]

**Review by zeroqn — APPROVED** [review]

**Pull request merged**